cf-file-scanner/config/example.conf

```diff
@@ -0,0 +1,18 @@
+# This is an example of config file needed by function.
+# You may copy-paste this for your own `local.conf` file.
+
+{
+
+  # When to refresh the database files, up to millisecond granularity.
+  # Unit need to match one of Java's `TimeUnit` enum variants.
+  # These are: DAYS, HOURS, MINUTES, SECONDS, MILLISECONDS.
+  "refreshTime": 1,
+  "refreshUnit": "HOURS",
+
+  # The GCP project. Used for both storage and pubsub.
+  "gcpProjectId": "string",
+
+  # The pubsub topic to publish scan results to.
+  "resultTopic": "string",
```

Comment:
You may want to have the topic defined via an env var for the GCF. This is usually the way to pass customization options to GCFs, because it's easier than having to modify a config file from CI/CD.

Reply:
I actually wanted to use a `env.yaml` file for these initially. However, these config values will be different for our different environment, so I'll need a way to render these on deploy. I guess I could put the `env.yaml` file into Secret Manager. But I think this setup is a little easier and more consistent to how Pepper does things. Also, I think it's a lot easier to pass in a config file than dealing with env variables during local development and testing.

Reply:
OK, that's fine for now. In general, it just makes function code less "portable" and more dependent on your particular deployment mechanism ;) Aka 12-factor apps :)

Reply:
I think the traditional approach is to store them in `.env` file and source that when you deploy, then you can just assign the values through the env files at deploy time instead of packaging the `.env` config itself with those. It also makes it easier to trace which function was deployed with which values - you can see those right in the Cloud Console!

Reply:
https://12factor.net/config

Reply:
I also was thinking more in terms of reusability of this GCF for other teams etc. If you're not looking to make this code useful to others, then it's fine to store config together with your code. I was more in favor of env vars because of reusability concern. So I think AppSec can then make a separate reusable implementation in our own repo, and treat this code as just useful for Pepper only.

Reply:
Okay, that does sound nice and convincing. I think I can get it to work with env vars. As a general question: what if we have a more secret-like config value, e.g. database url, are those still good to be set as env vars?

Reply:
For "true" secrets, definitely those should be set "out of band" (i.e. not stored in any config files..), usually by hand via Cloud Console for Secret Manager. This is what we've done for database connection strings in my previous team ;-)

Reply:
All set! I've switched to using env vars. It's really nice to see everyone about the function in one place in gcloud console. However, with env vars, I find myself doing something like this in the `deploy.sh` script:
```
source .env
VARS="GCP_PROJECT=$GCP_PROJECT"
VARS+=",RESULT_TOPIC=$RESULT_TOPIC"
...

gcloud ... --set-env-vars $VARS
```
because I need to build a string to pass to `gcloud functions deploy`. So instead of sourcing env file, I ended up rendering a `env.yaml` file from Secret Manager instead. The function itself and local development still uses env vars though.